share/wake/lib/rust_wake/cargo.wake

```diff
@@ -0,0 +1,45 @@
+# Copyright 2023 SiFive, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You should have received a copy of LICENSE.Apache2 along with
+# this software. If not, you may obtain a copy at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package rust
+
+target findCargoPath Unit =
+    require Some cargo_home = getenv "CARGO_HOME"
```

Comment:
This makes `CARGO_HOME` be taken in preference to `PATH`.
While unlikely, I think that would be unexpected behavior to shell users if they were inconsistent

Reply:
This is the whole point of CARGO_HOME though, it's used to override your cargo location in the cargo echo-system.

Reply:
To match the home crate we'd actually prefer $HOME/.cargo over $PATH: https://docs.rs/home/latest/home/fn.cargo_home.html and this is what the cargo crate recommends as well: https://doc.rust-lang.org/cargo/guide/cargo-home.html

(the home crate is a part of the cargo/rust-lang project and can also be used to find rustup/rustc in the default expected way)